components/Icon.tsx

```diff
@@ -0,0 +1,49 @@
+import clsx from "clsx";
+type Props = {
+  type: "home" | "search";
+  positionProps?: {
```

Comment:
you're overthinking it:

```suggestion
  className?: string
```